codes/quantum/oscillators/cat_code.yml

```diff
@@ -0,0 +1,97 @@
+code_id: cat_code
+physical: oscillators
+logical: qubits
+
+# Apostrophes are denoted by two apostrophe characters, i.e., ''
+# Code title is singular
+name: 'Cat code'
+
+# First reference with the code (optional).
+introduced: '\(|\alpha\rangle \pm |-\alpha\rangle\) cat code \cite{doi:10.1103/PhysRevA.59.2631}'
+
+# Anything applicable to a larger parent set of codes (see below) should go in
+# that entry instead of here.
+description: 'It is a continuous-variable code that encode a logical qubit into a harmonic oscillator. In phase space, cat code has rotation symmtry which can protect against dephasing noise. '
+
+protection: 'Protect against multi-photon loss, gain error and phase space rotation error up to \(\frac{\pi}{S+1}\).'
+
+# This field starts a list of specific labeled subfields; do not leave it
+# empty. If empty, comment out. Also, indentations are important!
+features:
+
+  # The fields below relate to specific properties of the code that are worth
+  # noting. In some cases (see below), there might be multiple pieces of
+  # information associated with a field (e.g., mutliple encoders/decoders,
+  # multiple facts about fault tolerance, etc.); in such cases, the field
+  # accepts a list of text entries, with one item per piece of information.
+  #
+  # Any fields below should be omitted if they do not apply.
+
+  rate: 'Encode a 2-dimensional logical Hilbert space into an infinite dimensional Hilbert space. A cat code with \(Z_{2(S+1)}\) rotation symmetry can protect against \(S\)-photon loss/gain errors and phase space rotation error up to \(\frac{\pi}{2(S+1)}\).'
+
+  # This field that starts a list of strings, indexed by a dash; do not leave it
+  # empty. If empty, comment out.
+  encoders: 
+    - 'Project a coherent state onto a subspace of occupation number modulo \(2(S+1)\). It is a superposition of coherent states around a circle.'
+    
+
+  #transversal_gates: 'Transversal ... gates \cite{doi:ok-paper}. Comment out if doesn''t apply.'
+
+  #general_gates:
+    #- ''
+
+  decoders:
+    - 'Measurement in the Fock basis. For a \(2(S+1)\) cat code, a number measurement returns outcome \(2(S+1)k\), if \(k\) is even, then it corresponds to logical 0 state; if \(k\) is odd, then it corresponds to logical 1 state.' 
+    #- '' 
+
+  #fault_tolerance:
+    #- 'Noise-model-preserving gadgets, noise-biased gates, fault-tolerant flag error correction'
+    #- 'Pieceable fault tolerance.'
+
+  #threshold: 
+    #- 'Threshold standard deviation of displacement errors is 11.2dB if Gaussian displacement error only applies to data and ancillas \cite{arxiv:another-paper}.'
+    #- '0.3\% error-correction threshold ... with noisy MWPM decoder \cite{doi:good-paper}.'
+
+  
+  # Propose a field that you feel is applicable to a large and interesting class of codes
+  #some_other_field:
+  # - 'detail'
+
+realizations:
+  # List and explain the different "domains" of realizations in list items.
```

Comment:
d=3 (S=2) realized in first work that achieved break-even error correction: 10.1038/nature18949. Please add + explain.